services/QuillLMS/client/app/bundles/Teacher/components/unit_templates/unit_template_activity_selector.tsx

```diff
@@ -0,0 +1,280 @@
+        (readabilitySearch === DEFAULT_READABILITY || (act.readability_grade_level && act.readability_grade_level === readabilitySearch)) &&
+        (selectedActivities.length === 0 || !selectedActivities.map(a => a.id).includes(act.id))
+      );
+    })
+  }
+
+  function currentPageActivities() {
+    return getFilteredActivities().slice(lowerBound(currentPage), upperBound(currentPage))
+  }
+
+  function findAndToggleParentActivity(activity) {
+    toggleParentActivity(activity)
+  }
+
+  function handleAddActivity(activity) {
+    findAndToggleParentActivity(activity)
+    const newSelectedActivities = selectedActivities.slice()
+    newSelectedActivities.push(activity)
+    setSelectedActivities(newSelectedActivities)
+  }
+
+  function handleRemoveActivity(activity) {
+    findAndToggleParentActivity(activity)
+    const newSelectedActivities = selectedActivities.filter(a => a.id !== activity.id)
+    setSelectedActivities(newSelectedActivities)
+  }
+
+  const tableHeaders = (
+    <tr className="ut-activities-headers">
+      <th className="ut-break">&nbsp;</th>
+      <th className="ut-break">&nbsp;</th>
+      <th className="ut-name-col">Name</th>
+      <th className="ut-flag-col">Flag</th>
+      <th className="ut-readability-col">Readability</th>
+      <th className="ut-ccss-col">CCSS</th>
+      <th className="ut-concept-col">Concept</th>
+      <th className="ut-tool-col">Tool</th>
+    </tr>
+  )
+
+
+  const activityRows = currentPageActivities().map((act) => {
+    return (
+      <UnitTemplateActivityDataRow
+        activity={act}
+        handleAdd={handleAddActivity}
+        handleRemove={handleRemoveActivity}
+        key={act.id}
+        type={UNSELECTED_TYPE}
+      />
+    )
+  })
+
+  function handleNameSearch(e) {
+    setNameSearch(e.target.value)
+  }
+
+  function handleDescriptionSearch(e) {
+    setDescriptionSearch(e.target.value)
+  }
+
+  function handleCCSSSearch(e) {
+    setCCSSSearch(e.target.value)
+  }
+
+  function handleConceptSearch(e) {
+    setConceptSearch(e.target.value)
+  }
+
+  function handleActivityPacksSearch(e) {
+    setActivityPacksSearch(e.target.value)
+  }
+
+  function handleFlagSearch(e) {
+    setFlagSearch(e.target.value)
+  }
+
+  function handleToolSearch(e) {
+    setToolSearch(e.target.value)
+  }
+
+  function handleReadabilitySearch(e) {
+    setReadabilitySearch(e.target.value)
+  }
+
+  function updateOrder(sortInfo) {
+    const sortedIds = sortInfo.map(s => s.key)
+    const newOrderedActivities = sortedIds.map((s) => activities.find((a) => a.id === parseInt(s)))
+    setSelectedActivities(newOrderedActivities)
+
+    const selectedParentActivities = newOrderedActivities.map(act => parentActivities.find(a => a.id === act.id))
+    setParentActivities(selectedParentActivities)
+  }
+
+  function selectedActivitiesTable() {
+    const fullSelectedActivities = activities.length > 0 ? selectedActivities.map((act) => activities.find(a => act.id === a.id)) : []
```

Comment:
Tiny but since 0 is falsy, you can just do `activities.length ? selectedActivities...`